tricycle_steering_controller/src/tricycle_steering_controller.cpp

```diff
@@ -60,24 +60,26 @@ bool TricycleSteeringController::update_odometry(const rclcpp::Duration & period
   }
   else
   {
-    const double rear_right_wheel_value = state_interfaces_[STATE_TRACTION_RIGHT_WHEEL].get_value();
-    const double rear_left_wheel_value = state_interfaces_[STATE_TRACTION_LEFT_WHEEL].get_value();
+    const double traction_right_wheel_value =
+      state_interfaces_[STATE_TRACTION_RIGHT_WHEEL].get_value();
+    const double traction_left_wheel_value =
+      state_interfaces_[STATE_TRACTION_LEFT_WHEEL].get_value();
     const double steer_position = state_interfaces_[STATE_STEER_AXIS].get_value();
     if (
-      !std::isnan(rear_right_wheel_value) && !std::isnan(rear_left_wheel_value) &&
-      !std::isnan(steer_position))
+      std::isfinite(traction_right_wheel_value) && std::isfinite(traction_left_wheel_value) &&
```

Comment:
What does `traction` mean in this context?

We have front and rear wheel actuated tricycles.

Reply:
steering_controller lib does implement tricycles with traction on the other axle as the steering wheel (different than the earlier tricycle_controller, assuming steering+traction no the same wheel), but with front or rear steering (subsequently, rear or front traction)

Here, the old variable name was `rear_right/left_wheel_value` independent of the `params_.front_steering` value. See the `state_interface_configuration()` method, where depending on the `params_.front_steering` the front or rear wheel names are exported as state interfaces, which is read hear as `state_interfaces_[STATE_TRACTION_RIGHT_WHEEL].get_value()`. I just renamed the variable to be valid for both configurations.

Or have I misunderstood something?

Reply:
> steering_controller lib does implement tricycles with traction on the other axle as the steering wheel (different than the earlier tricycle_controller, assuming steering+traction on the same wheel)

Ah, this might be part of the confusion. But it only supports a subset of tricycles then. In that case this variable is in line with that.

Reply:
exactly. I have plans to include the "old" tricycle controller into the steering lib, but step by step